### ISSUE TITLE
Error on Android when change screen orientation

### ISSUE THREAD

**Issue by Pip3r4o** [NativeScript/NativeScript#4966]
_From @dgmaar on October 18, 2017 18:27_

## Please, provide the details below:
Hi, I'm getting this error when switch app orientation, I noticed that this happen when property clearHistory = true when navigating pages

An uncaught Exception occurred on "main" thread.
java.lang.RuntimeException: Unable to start activity ComponentInfo{mx.sicar.sicarapp/com.tns.NativeScriptActivity}: com.tns.NativeScriptException: 
Calling js method onCreate failed

Error: Could not find a page for fragment0[0].
File: "file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/frame.js, line: 584, column: 18

StackTrace: 
	Frame: function:'findPageForFragment', file:'file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/frame.js', line: 385, column: 15
	Frame: function:'FragmentCallbacksImplementation.onCreate', file:'file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/frame.js', line: 465, column: 13
	Frame: function:'FragmentClass.onCreate', file:'file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/fragment.js', line: 21, column: 25
	Frame: function:'ActivityCallbacksImplementation.onCreate', file:'file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/frame.js', line: 584, column: 19
	Frame: function:'NativeScriptActivity.onCreate', file:'file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/activity.js', line: 20, column: 25

	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2572)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2654)
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4317)
	at android.app.ActivityThread.-wrap15(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1494)
	at android.os.Handler.dispatchMessage(Handler.java:111)
	at android.os.Looper.loop(Looper.java:207)
	at android.app.ActivityThread.main(ActivityThread.java:5728)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:888)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:749)
Caused by: com.tns.NativeScriptException: 
Calling js method onCreate failed

Error: Could not find a page for fragment0[0].
File: "file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/frame.js, line: 584, column: 18

StackTrace: 
	Frame: function:'findPageForFragment', file:'file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/frame.js', line: 385, column: 15
	Frame: function:'FragmentCallbacksImplementation.onCreate', file:'file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/frame.js', line: 465, column: 13
	Frame: function:'FragmentClass.onCreate', file:'file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/fragment.js', line: 21, column: 25
	Frame: function:'ActivityCallbacksImplementation.onCreate', file:'file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/frame.js', line: 584, column: 19
	Frame: function:'NativeScriptActivity.onCreate', file:'file:///data/data/package/files/app/tns_modules/tns-core-modules/ui/frame/activity.js', line: 20, column: 25

	at com.tns.Runtime.callJSMethodNative(Native Method)
	at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1084)
	at com.tns.Runtime.callJSMethodImpl(Runtime.java:966)
	at com.tns.Runtime.callJSMethod(Runtime.java:953)
	at com.tns.Runtime.callJSMethod(Runtime.java:937)
	at com.tns.Runtime.callJSMethod(Runtime.java:929)
	at com.tns.NativeScriptActivity.onCreate(NativeScriptActivity.java:13)
	at android.app.Activity.performCreate(Activity.java:6270)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1113)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2519)
	... 10 more

### Did you verify this is a real problem by searching [Stack Overflow]
Yes

### Please provide the following version numbers that your issue occurs with:
│ nativescript     │ 3.2.1           │ 3.2.1       
│ tns-core-modules │ 3.2.0           │ 3.2.0        
│ tns-android      │ 3.2.0           │ 3.2.0         
│ tns-ios          │ 3.2.0           │ 3.2.0        

plugins

    "base-64": "^0.1.0",
    "crypto-js": "^3.1.9-1",
    "nativescript-apiclient": "^1.13.0",
    "nativescript-barcodescanner": "^2.1.1",
    "nativescript-drop-down": "^3.1.0",
    "nativescript-email": "^1.4.0",
    "nativescript-ethernet-printer": "file:nativescript-ethernet-printer",
    "nativescript-fingerprint-auth": "^3.0.1",
    "nativescript-https": "^1.0.1",
    "nativescript-iqkeyboardmanager": "^1.0.1",
    "nativescript-miniprinter-ios": "file:pluginM06/nativescript-miniprinter-ios",
    "nativescript-orientation": "^1.6.1",
    "nativescript-pdf-view": "^2.0.1",
    "nativescript-printer": "file:plugin/nativescript-printer",
    "nativescript-pushservice": "file:nativescript-pushservice",
    "nativescript-slides": "^2.2.8",
    "nativescript-telerik-ui": "^3.0.0-2017.6.28.1",
    "nativescript-timedatepicker": "^1.0.0",
    "nativescript-xml2js": "^0.5.2",
    "tns-core-modules": "^3.2.0",
    "utf8": "^2.1.2"

### Did the error happen while the app was executing? (runtime error)
Yes
Please provide us with the log by following these steps:
* Add `__enableVerboseLogging()` in the `app.js` file.
* Run command: `adb logcat`.
* Attach the content of the log to the Issue. You can use [pastebin](http://pastebin.com/).
-->

### Please tell us how to recreate the issue in as much detail as possible.

_Copied from original issue: NativeScript/android-runtime#862_

**Comment by tsonevn** [comment]
Hi @dgmaar,

I reviewed your case and was able to recreate the similar problem on my side.  In my investigation, I found that the issue is related to the `nativescript-orientation` plugin and it is not something that depends on tns-core-modules. 
Regarding that, I would suggest to contact the plugin's author and to report this issue in [its repository](https://github.com/NathanaelA/nativescript-orientation).

**Comment by dgmaar** [comment]
Thank you very much for reply! I'll report it to author

**Comment by NathanaelA** [comment]
@tsonevn - How did you determine it was ns-orientation?   I just tested the new version of ns-orientation in a demo app; with navigation (& clearHistory: true) and I could not see any failures.   The callstack above also doesn't mention ns-orientation in it.   So I'm a bit confused why you think it is ns-orientation, or do you have some way to cause this failure?

**Comment by irthos** [comment]
i'm not sure it has to do with orientation, although I'm also using this plugin. Get the same error when commenting out the import and use of screen-orientation.

**Comment by NathanaelA** [comment]
@tsonevn - I use NS-orientation with multiple apps; and have never seen this failure.  I would be very interested in knowing how you duplicated it as it is my plugin.   In addition the callstack doesn't indicate anything to do with ns-orientation, so I'm not sure why you think ns-orientation caused this issue.

**Comment by tsonevn** [comment]
Hi @NathanaelA,
I will review the issue again a little bit late today and will provide more info about the case.

**Comment by tsonevn** [comment]
Hi @NathanaelA 
I was able to recreate this issue while using `nativescript-orientation` version 1.6.1 and the latest version of  `tns-core-modules` 3.3.0. It looks like that the old plugin's version is not compatible with the new NativeScript 3. This was the reason, why I redirect the customer to the plugins repository. 
However, while using the latest version of the plugin `2.1.0`, the project crash is no longer reproduced.
 
Also attaching the project that I used for debugging.
[Archive.zip](https://github.com/NativeScript/NativeScript/files/1501403/Archive.zip)
